cpp/src/arrow/sparse_tensor.cc

```diff
@@ -623,6 +795,75 @@ SparseCSRIndex::SparseCSRIndex(const std::shared_ptr<Tensor>& indptr,
 
 std::string SparseCSRIndex::ToString() const { return std::string("SparseCSRIndex"); }
 
+// ----------------------------------------------------------------------
+// SparseCSFIndex
+
+namespace {
+
+inline Status CheckSparseCSFIndexValidity(const std::shared_ptr<DataType>& indptr_type,
+                                          const std::shared_ptr<DataType>& indices_type,
+                                          const int64_t num_indptrs,
+                                          const int64_t num_indices,
+                                          const std::vector<int64_t>& indptr_shape,
+                                          const std::vector<int64_t>& indices_shape,
+                                          const int64_t axis_order_size) {
+  if (!is_integer(indptr_type->id())) {
+    return Status::Invalid("Type of SparseCSFIndex indptr must be integer");
+  }
+  if (!is_integer(indices_type->id())) {
+    return Status::Invalid("Type of SparseCSFIndex indices must be integer");
+  }
+  if (num_indptrs + 1 != num_indices) {
+    return Status::Invalid(
+        "SparseCSFIndex length indices must be equal to length inptrs plus one.");
+  }
+  if (axis_order_size != num_indices) {
+    return Status::Invalid(
+        "SparseCSFIndex length of indices must be equal number of dimensions.");
+  }
+  return Status::OK();
+}
+
+}  // namespace
+
+Result<std::shared_ptr<SparseCSFIndex>> SparseCSFIndex::Make(
+    const std::shared_ptr<DataType>& indptr_type,
+    const std::shared_ptr<DataType>& indices_type,
+    const std::vector<int64_t>& indices_shapes, const std::vector<int64_t>& axis_order,
+    std::vector<std::shared_ptr<Buffer>> indptr_data,
+    std::vector<std::shared_ptr<Buffer>> indices_data) {
+  int64_t ndim = axis_order.size();
+  std::vector<std::shared_ptr<Tensor>> indptr(ndim - 1);
+  std::vector<std::shared_ptr<Tensor>> indices(ndim);
+
+  for (int64_t i = 0; i < ndim - 1; ++i)
+    indptr[i] = std::make_shared<Tensor>(indptr_type, indptr_data[i],
+                                         std::vector<int64_t>({indices_shapes[i] + 1}));
+
+  for (int64_t i = 0; i < ndim; ++i)
+    indices[i] = std::make_shared<Tensor>(indices_type, indices_data[i],
+                                          std::vector<int64_t>({indices_shapes[i]}));
+
```

Comment:
Why not validate parameters here by `CheckSparseCSFIndexValidity`?

Reply:
Would it not be checked when the [construcor](https://github.com/apache/arrow/blob/bc0e27831a6b3f4ac27247149a660db7dcc60920/cpp/src/arrow/sparse_tensor.cc#L860) is called?

Reply:
When the validation in the constructor fails, the program is aborted by `ARROW_CHECK`.
To make `Make` function safe, it is necessary to calling the validation in this function.

Reply:
Got it! Added it now.